### PR TITLE
remove deprecated broadcast in test file

### PR DESCRIPTION
**Pull request by KlausC**
this change in a test file remove a test block when StaticArrays@1.0 is used

### CODE REVIEW

File: test/interface/static_array_tests.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ ode = ODEProblem(f, u0, (0.,1.))
 sol = solve(ode, Euler(), dt=1e-2)
 sol = solve(ode, Tsit5())
 
-u0 = fill(zero(SVector{2,Float64}), 2) .+ 1
+u0 = fill(zero(SVector{2,Float64}), 2)
 u0[1] = ones(SVector{2,Float64}) .+ 1
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:13]: This leads to a different initialization than before, it seems? Maybe it would be easier to just use:

```suggestion
u0 = [fill(2, SVector{2,Float64}), ones(SVector{2,Float64})]
```

Comment (KlausC) [reply test/interface/static_array_tests.jl:13]: the values may be irrelevant in this test.

Comment (devmotion) [reply test/interface/static_array_tests.jl:13]: A solution with a component that is zero constantly did not seem useful but probably it does not matter here and I noticed that the test with `MVector` above also uses a zero component. However, even in this case I'd say the following initialization
```suggestion
u0 = [fill(2, SVector{2,Float64}), zeros(SVector{2,Float64})]
```
is more natural (similarly, the example above could be changed).

Comment (KlausC) [reply test/interface/static_array_tests.jl:13]: I think the " .+ 1" it was just a typo, which did not any harm as long as `+(::SArray, ::Number)` was defined.

Comment (devmotion) [reply test/interface/static_array_tests.jl:13]: No, it was completely intentional. The tests were added a long time ago, I assume when only `zeros` but not `ones` was defined for StaticArrays. Initially, it was even just `fill(...) + 1` (without the broadcast) since this was valid syntax back then.

**Comment by ChrisRackauckas** [comment]
The values shouldn't matter too much, but making it all zeros makes it a bit weaker. It should have a non-zero initialization and check the final solution isn't pure zero or the same as the initial condition, as that would be a sign of aliasing issues.

**Comment by ChrisRackauckas** [comment]
We should probably strengthen it just a little to add the test for non-zero-ness in the end.

**Comment by KlausC** [comment]
adapted recommendations also to other cases in the surrounding.

**Review by ChrisRackauckas — APPROVED** [review]

File: test/interface/static_array_tests.jl [hunk 2]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
 using OrdinaryDiffEq
 using RecursiveArrayTools
 
-u0 = fill(zero(MVector{2,Float64}), 2)
-u0[1] = ones(MVector{2,Float64}) .+ 1
+u0 = [fill(2, MVector{2,Float64}), ones(MVector{2,Float64})]
 f = (du,u,p,t) -> du .= u
 ode = ODEProblem(f, u0, (0.,1.))
 sol = solve(ode, Euler(), dt=1e-2)
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:9]: Just a minor suggestion (feel free to ignore it):
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 3]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, Tsit5())
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:11]: Same here:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 4]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 
-u0 = [fill(2, SVector{2,Float64}), zeros(SVector{2,Float64})]
+u0 = [fill(2, SVector{2,Float64}), ones(SVector{2,Float64})]
 ode = ODEProblem(f, u0, (0.,1.))
 sol = solve(ode, Euler(), dt=1e-2)
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:16]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 5]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, Tsit5())
-
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:18]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 6]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, SSPRK22(), dt=1e-2)
-
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:20]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 7]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 
 u0 = ones(MVector{2,Float64})
 ode = ODEProblem(f, u0, (0.,1.))
 sol = solve(ode, Euler(), dt=1e-2)
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:25]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 8]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, Tsit5(), dt=1e-2)
-
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:27]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 9]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 
 u0 = ones(SVector{2,Float64})
 f = (u,p,t) -> u
 ode = ODEProblem(f, u0, (0.,1.))
 sol = solve(ode, Euler(), dt=1e-2)
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:33]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 10]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 integrator = init(ode, ImplicitEuler())
 @test OrdinaryDiffEq.get_W(integrator.cache.nlsolver) isa StaticArrays.LU
 sol = solve(ode, ImplicitEuler())
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:37]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 11]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, ImplicitEuler(nlsolve=NLAnderson()))
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:39]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

File: test/interface/static_array_tests.jl [hunk 12]
```diff
@@ -2,36 +2,43 @@ using StaticArrays, Test
...
 sol = solve(ode, Tsit5(), dt=1e-2)
+@test !any(iszero.(sol(1.0))) && !any(sol(1.0) .== u0)
```
Comment (devmotion) [inline test/interface/static_array_tests.jl:41]:
```suggestion
@test !any(iszero, sol(1.0)) && !any(sol(1.0) .== u0)
```

Comment (KlausC) [reply test/interface/static_array_tests.jl:41]:
```
julia> @btime any($iszero, $sol(1.0))
  2.886 μs (34 allocations: 992 bytes)
false

julia> @btime any($iszero.($sol(1.0)))
  2.968 μs (36 allocations: 1.09 KiB)
```
